dropshot/src/server.rs

```diff
@@ -154,6 +154,16 @@ impl<C: ServerContext> HttpServerStarter<C> {
         let server = builder.serve(make_service);
         info!(app_state.log, "listening");
 
+        if let Err(e) = crate::register_probes() {
+            warn!(
```

Comment:
I think this should be an error.  (I almost want to say it should be fatal.  I've been burned by things silently failing to enable DTrace support.  And if it was built for it, there's no reason this should fail at runtime, right?  But I imagine this sounds extreme.)

Reply:
Ahh, wait you mean return an error from the start method. Yes, that seems totally fine to me.

Reply:
I did actually mean an "error" level log message.  I could also see letting the caller decide.  I'm curious what @ahl thinks.

Reply:
To @davepacheco 's earlier point: it would suck to have this fail and not notice. While it's true, @bnaecker, that there's an `ioctl(2)`, I wouldn't describe this as I/O per se.

I think this raises the question: "how could this fail transiently?" I think only if the system was low on memory and it could not allocate the necessary kernel data structures needed to store the provider state.  I can't think of another scenario where registration would fail transiently.

In that situation what would we want? Perhaps that answer would vary from service to service. If the system is already in a rocky state, perhaps we'd rather have, say, a service up that can tell us that we're low on memory. Unfortunately I could see either option (panic!() or ignore) biting us in various situations.

Reply:
It sounds like there isn't an option that doesn't trade some things off. We can emit an error message to the log, no problem. Returning an `Err` seems OK, but there's no way for the caller to indicate that they don't care about such a situation and want to start the server regardless. (Not that we'd want to do that, but someone else might.) If we panic, that's also not great for obvious reasons.

I think the right thing here is to not have the server call `register_probes`, but to let the caller do it via that public function. It's a bit risky, since they have to read the documentation saying that they need to call this, but I don't see a better way. It seems overall worse to, for example, include an argument in the `start()` method saying how to handle this situation -- just let the caller do it, right?

Reply:
I've updated this in b52dc29, which requires that the caller register probes and gives them the option of how to handle failure. The examples now all do `dropshot::register_probes().expect("Failed to register DTrace probes")`. Let me know if y'all have further thoughts on this.

Reply:
Change of plan -- I've now stored the result of registration in a field of the `HttpServer`. Probes are always registered if possible (no need to call `register_probes()` manually), and callers can still decide how to handle failures.